xonsh/environ.py

```diff
@@ -690,7 +690,8 @@ def yield_executables_windows(directory, name):
     extensions = builtins.__xonsh_env__.get('PATHEXT')
     for a_file in os.listdir(directory):
         base_name, ext = os.path.splitext(a_file)
-        if name == base_name and ext.upper() in extensions:
+
+        if (name == base_name or a_file == name) and ext.upper() in extensions:
```

Comment:
I think we should have this change regardless of this patch. This essentially fails to find executables if you provide their full name (with the extension).